### PR TITLE
WFLY-15127 Drop JGroups subsystem model versions/transformers prior to 8.0.0

### PR DESCRIPTION
**Pull request by pferraro**
https://issues.redhat.com/browse/WFLY-15127

### CODE REVIEW

**Comment by pferraro** [comment]
@rhusar Please review

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/ResourceDefinitionProvider.java [hunk 1]
```diff
@@ -44,5 +44,8 @@
      * @param parent the builder of the parent resource
      * @param version the version to which to transform
      */
-    void buildTransformation(ResourceTransformationDescriptionBuilder parent, ModelVersion version);
+    @Deprecated
```
Comment (rhusar) [inline clustering/common/src/main/java/org/jboss/as/clustering/controller/ResourceDefinitionProvider.java:47]: Ideally this should be accompanied by corresponding Javadoc update explaining what to use instead.

https://docs.oracle.com/javase/7/docs/technotes/guides/javadoc/deprecation/deprecation.html#javadoc_tag

File: clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/JGroupsExtensionTransformerRegistration.java [hunk 2]
```diff
@@ -22,16 +22,16 @@
...
 import java.util.EnumSet;
 
 import org.jboss.as.controller.ModelVersion;
 import org.jboss.as.controller.transform.ExtensionTransformerRegistration;
 import org.jboss.as.controller.transform.SubsystemTransformerRegistration;
+import org.jboss.as.controller.transform.description.TransformationDescription;
 import org.kohsuke.MetaInfServices;
 
 /**
+ * Registration for JGroups subsystem tranformers.
```
Comment (rhusar) [inline clustering/jgroups/extension/src/main/java/org/jboss/as/clustering/jgroups/subsystem/JGroupsExtensionTransformerRegistration.java:34]: s/tranformers/transformers

**Review by rhusar — APPROVED** [review]
Looks good to me.

File: clustering/common/src/main/java/org/jboss/as/clustering/controller/ResourceDefinitionProvider.java (hunk 1 above)
Comment (pferraro) [reply clustering/common/src/main/java/org/jboss/as/clustering/controller/ResourceDefinitionProvider.java:47]: Normally, yes, but this isn't a public module.

**Pull request merged**